discovery/data_sharing/v0.1/data_sharing.md

```diff
@@ -0,0 +1,297 @@
+  * `shares`: Number of shares in the given timeframe, defaults to `3`
+  * `likes`: Number of likes in the given timeframe, defaults to `3`
+  * `replies`: Number of replies in the given timeframe, defaults to `3`
+
+Example object:
+
+```json
+{
+  "category": "content",
+  "subscriptionType": "trends",
+  "maxBatchSize": 10,
+  "threshold": {
+    "timeframe": 15,
+    "shares": 3
+  }
+}
+```
+
+The fediverse server MUST validate the request. If it is invalid it MUST
+return an HTTP status code `422` (Unprocessable Content).
+
+If it is valid the response from the fediverse server MUST include an
+HTTP status code `201` (Created) and a JSON object including the `id` of
+the generated subscription. The latter can be used to unsubscribe later.
+
+Example response object:
+
+```json
+{
+  "subscriptionId": 3446
+}
+```
+
+To unsubscribe the FASP can make an HTTP `DELETE` call to the
+`/data_sharing/v0/event_subscriptions/:id` endpoint, where `:id` is replaced
+with the ID received when the original subscription was created.
+
+Example call:
+
+```http
+DELETE /data_sharing/v0/event_subscriptions/3446
+```
+
+The response MUST be an HTTP status code `200` (OK).
+
+### Requesting Historic Content / Backfilling (FASP => Fediverse Server)
+
+To request historic content to index a FASP can make an HTTP `POST` call
+to the `/data_sharing/v0/backfill_requests` endpoint.
+
+Example call:
+
+```http
+POST /data_sharing/v0/backfill_requests
+```
+
+The request body MUST contain a JSON object with the following keys:
+
+* `category`: One of `content`, `account`. This is the category
+  of objects the FASP is interested in.
+* `maxCount`: An integer representing the maximum number of objects the
+  FASP would like to receive.
+* `cursor`: Optional. An opaque token received from the fediverse server
+  in a previous request. MUST be used by the fediverse server to
+  determine results that have not already been sent to the FASP.
+
+Example object:
+
+```json
+{
+  "category": "content",
+  "maxCount": 100,
+  "cursor": "1541815103606536472"
+}
+```
+
+The response from the fediverse server MUST include an HTTP status code
+`201` (Created) and a JSON object including the `id` of the generated
+backfill request.
+
+Example response object:
+
+```json
+{
+  "backfillRequestId": 672
+}
+```
+
+### Sending Content (Fediverse Server => FASP)
+
+Both the occurrence of an event that a FASP has subscribed to and the
+fulfillment of a backfill request, MUST be announced to the FASP by the
+fediverse server with an HTTP `POST` call to the FASP's
+`/data_sharing/v0/announcements` endpoint.
+
+Example call:
+
+```http
+POST /data_sharing/v0/announcements
+```
+
+The request body MUST include a JSON object including the keys
+`source`, `objectType` and `objectUris`.
+
+* `source` lets the FASP know in reply to which of its request this
+  announcement is sent. It MUST include an object with either a
+  `subscriptionId` or a `backfillRequestId`.
```

Comment:
Would once again recommend:
```
source: {
  type: "subscription" | "backfill_request"
  id: "1234567"
}
```

Reply:
I think it would be more consistent this way:

```json
"source": {
  "subscription": {
    "id": "132542"
  }
}
```

I just pushed a change to that effect but am still  open to alternatives.